airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -488,7 +498,9 @@ def create_pod_request_obj(self) -> k8s.V1Pod:
             pod = secret.attach_to_pod(pod)
         if self.do_xcom_push:
             self.log.debug("Adding xcom sidecar to task %s", self.task_id)
-            pod = xcom_sidecar.add_xcom_sidecar(pod)
+            pod = xcom_sidecar.add_xcom_sidecar(pod,
+                                                self.xcom_sidecar_container
+                                                or xcom_sidecar.PodDefaults.SIDECAR_CONTAINER)
```

Comment:
I wonder if instead of being all or nothing, we use `PodGenerator.reconcile_pods` to merge it? You sorta mention it already in the docstring by saying "modification of `PodDefaults.SIDECAR_CONTAINER`", but maybe we should do it automatically?

e.g, similar to how we handle `full_pod_spec` for the main container in [create_pod_request_obj()](https://github.com/apache/airflow/blob/eeb45733e70442122c50e99a5ac5fac38c96d080/airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py#L427)